node/bft/src/helpers/proposal_cache.rs

```diff
@@ -144,15 +159,20 @@ impl<N: Network> FromBytes for ProposalCache<N> {
         };
         // Deserialize `signed_proposals`.
         let signed_proposals = SignedProposals::read_le(&mut reader)?;
+        // Read the number of pending certificates.
+        let num_certificates = u32::read_le(&mut reader)?;
```

Comment:
We need to safety check this number so it does not overallocate.

Reply:
Resolved in https://github.com/AleoHQ/snarkOS/pull/3247/commits/7ed91da2e094ed6e47ad7dcffa1a2b2e1af1e593